src/components/ChartJsWrapper/index.jsx

```diff
@@ -33,50 +34,70 @@ const ChartJsWrapper = ({
   type,
   chartHeight,
   spinnerSize,
-  missingDataError = false,
-}) => {
-  let showError;
+}) =>
+  (() => {
+    if (!data) {
+      return (
+        <div
+          style={{
+            lineHeight: spinnerSize,
+            textAlign: 'center',
+            width: spinnerSize,
+          }}>
+          <CircularProgress />
+        </div>
+      );
+    }
+
+    let error = null;
 
-  if (data && missingDataError) {
-    showError = data.datasets.every(dataset => {
-      const latestDataDate = new Date(dataset.data[dataset.data.length - 1].x);
+    data.datasets.forEach(dataset => {
+      const latestDataDate = new Date(
+        frum(dataset.data)
+          .sort('x')
+          .last().x
+      );
       const currentDate = new Date(); // get current date
       const timeDifference = Math.abs(
         currentDate.getTime() - latestDataDate.getTime()
       );
       const daysDifference = Math.ceil(timeDifference / (1000 * 3600 * 24));
 
-      return daysDifference > 3; // if days are more than 3 then show error
+      if (daysDifference > 3) {
+        error = new Error(
+          'This item has been missing data for at least 3 days.'
+        );
+      }
     });
-  }
 
-  return data ? (
-    <div className={classes.chartContainer}>
-      {showError && (
-        <CriticalErrorMessage
-          className={classes.errorPanel}
-          error="This item has been missing data for at least 3 days."
+    if (error) {
+      return (
+        <div className={classes.chartContainer}>
+          <ErrorMessage error={error}>
+            {title && <h2>{title}</h2>}
+            <Chart
+              type={type}
+              data={data}
+              height={chartHeight}
+              options={generateOptions(options)}
+            />
+          </ErrorMessage>
+        </div>
+      );
+    }
+
+    return (
+      <div className={classes.chartContainer}>
+        {title && <h2>{title}</h2>}
+        <Chart
+          type={type}
+          data={data}
+          height={chartHeight}
+          options={generateOptions(options)}
         />
-      )}
-      {title && <h2>{title}</h2>}
-      <Chart
-        type={type}
-        data={data}
-        height={chartHeight}
-        options={generateOptions(options)}
-      />
-    </div>
-  ) : (
-    <div
-      style={{
-        lineHeight: spinnerSize,
-        textAlign: 'center',
-        width: spinnerSize,
-      }}>
-      <CircularProgress />
-    </div>
-  );
-};
+      </div>
+    );
+  })();
```

Comment:
Why did you turn it into a function call?

Reply:
The `(() => {...})()` pattern turns a block of code (`...`) into an expression.  This allows us to make the code more readable:

Instead of trinary expressions and Boolean `&&` tricks:

```javascript
data ? ( showError && <error> ) <data> : <spinner>
```

we can say 

```javascript
(()=>{
    if (showError) return <error>;
    if (!data) return <spinner>;
    return <data>;
})();
```

which is much nicer when the `<components>` are complicated